src/main/java/LittleHarppy/Parser.java

```diff
@@ -0,0 +1,175 @@
+package LittleHarppy;
+
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+import file.Storage;
+import LittleHarppy.exception.LittleHarppyException;
+
+import java.util.ArrayList;
+
+/**
+ * Responsible for parsing user commands and processing the relevant task actions.
+ * This class determines which command was entered by the user and calls the corresponding method
+ * to handle the action.
+ */
+public class Parser {
+
+    /**
+     * Processes a given user command and executes the corresponding task.
+     *
+     * @param input the user input
+     * @param tasks the task list to manipulate
+     * @param ui the UI to display information to the user
+     * @param storage the storage for saving and loading tasks
+     * @return false if the user typed 'bye', otherwise true
+     */
+    public boolean processCommand(String input, TaskList tasks, Ui ui, Storage storage) {
+        try {
+            if (input.equalsIgnoreCase("bye")) {
+                return false;  // Ends chatbot loop
+            } else if (input.equalsIgnoreCase("list")) {
+                tasks.listTasks();
+            } else if (input.startsWith("find ")) {
+                String keyword = input.substring(5).trim(); // Extract the keyword
+                ArrayList<Task> matchingTasks = tasks.findTasksByKeyword(keyword);
+                ui.showFoundTasks(matchingTasks); // Show the found tasks
+            } else if (input.startsWith("mark ")) {
+                markTask(input, tasks, ui);
+            } else if (input.startsWith("unmark ")) {
+                unmarkTask(input, tasks, ui);
+            } else if (input.startsWith("delete ")) {
+                deleteTask(input, tasks, ui);
+            } else if (input.startsWith("todo ")) {
+                addTodo(input, tasks, ui);
+            } else if (input.startsWith("deadline ")) {
+                addDeadline(input, tasks, ui);
+            } else if (input.startsWith("event ")) {
+                addEvent(input, tasks, ui);
+            } else {
+                throw LittleHarppyException.invalidInputFormat();
+            }
+
+            // Save tasks after every modification
+            storage.save(tasks.getTasks());
+
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+        return true;
+    }
+
+    // Helper methods for different task actions
+
+    /**
+     * Adds a Todo task to the task list.
+     * @param input the user input for creating a Todo task
+     * @param tasks the task list to modify
+     * @param ui the UI to display output
+     * @throws LittleHarppyException if the Todo task description is empty
+     */
+    private void addTodo(String input, TaskList tasks, Ui ui) throws LittleHarppyException {
+        String taskDescription = input.substring(5).trim();
+        if (taskDescription.isEmpty()) {
```

Comment:
Good job with making the [happy path prominent](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent).